client/services/SolutionHandlerServiceSpec.js

```diff
@@ -222,5 +235,68 @@ describe('SolutionHandlerService', function() {
         });
       });
     });
+
+    describe("prereqCheckFailures", function() {
+      it('should be correctly handled if missing starter code', function(done) {
+        SolutionHandlerService.processSolutionAsync(
+          orderedTasks, starterCode, '',
+          auxiliaryCode, 'python'
+        ).then(function(feedback) {
+          expect(feedback.isAnswerCorrect()).toEqual(false);
+          expect(feedback.getParagraphs()[0].getContent()).toEqual([
+            'It looks like you deleted or modified the starter code!  Our ',
+            'evaluation program requires the function names given in the ',
+            'starter code.  You can press the \'Reset Code\' button to start ',
+            'over.  Or, you can copy the starter code below:'
+          ].join(''));
+          expect(feedback.getParagraphs()[1].getContent()).toEqual(starterCode);
+          done();
+        });
+      });
+
+      it('should be correctly handled if has bad import', function(done) {
+        var studentCode = [
+          'import pandas',
+          'def mockMainFunction(input):',
+          '    return True'
+        ].join('\n');
+
+        SolutionHandlerService.processSolutionAsync(
+          orderedTasks, starterCode, studentCode,
+          auxiliaryCode, 'python'
+        ).then(function(feedback) {
+          expect(feedback.isAnswerCorrect()).toEqual(false);
+          expect(feedback.getParagraphs()[0].getContent()).toEqual([
+            "It looks like you're importing an external library. However, the ",
+            'following libraries are not supported:\n'
+          ].join(''));
+          expect(feedback.getParagraphs()[1].getContent()).toEqual('pandas');
+          expect(feedback.getParagraphs()[2].getContent()).toEqual(
+            'Here is a list of libraries we currently support:\n');
+          expect(feedback.getParagraphs()[3].getContent()).toEqual(
+            SUPPORTED_PYTHON_LIBS.join(', '));
+          done();
+        });
+      });
+    });
+
+    describe('potentialSyntaxError', function() {
+      it('should correctly handle a possible syntax error', function(done) {
```

Comment:
I was using that word choice because of the phrasing within the code and its variables, but I see what you mean. Done!